qcodes/dataset/data_set.py

```diff
@@ -708,21 +733,41 @@ def get_data(self,
             be of the datatypes stored in the database (numeric, array or
             string)
         """
-        valid_param_names = []
-        for maybeParam in params:
-            if isinstance(maybeParam, str):
-                valid_param_names.append(maybeParam)
-                continue
-            else:
-                try:
-                    maybeParam = maybeParam.name
-                except Exception as e:
-                    raise ValueError(
-                        "This parameter does not have  a name") from e
-            valid_param_names.append(maybeParam)
-        data = get_data(self.conn, self.table_name, valid_param_names,
+        valid_param_names = self._validate_parameters(*params)
+        return get_data(self.conn, self.table_name, valid_param_names,
+                        start, end)
+
+    def get_parameter_data(
+            self,
+            *params: Union[str, ParamSpec, _BaseParameter],
+            start: Optional[int] = None,
+            end: Optional[int] = None) -> Dict[str, numpy.ndarray]:
```

Comment:
The method fails rather loudly if no parameters are supplied. (The same is the case for get_data) but never the less perhaps we should test for this

```
ERROR:qcodes.dataset.sqlite_base:Rolling back due to unhandled exception
Traceback (most recent call last):
  File "c:\users\jenielse\source\repos\qcodes\qcodes\dataset\sqlite_base.py", line 793, in atomic
    yield conn
  File "c:\users\jenielse\source\repos\qcodes\qcodes\dataset\sqlite_base.py", line 756, in atomic_transaction
    c = transaction(conn, sql, *args)
  File "c:\users\jenielse\source\repos\qcodes\qcodes\dataset\sqlite_base.py", line 733, in transaction
    c.execute(sql)
sqlite3.OperationalError: near "FROM": syntax error

---------------------------------------------------------------------------
OperationalError                          Traceback (most recent call last)
c:\users\jenielse\source\repos\qcodes\qcodes\dataset\sqlite_base.py in atomic(conn)
    792             conn.cursor().execute('BEGIN')
--> 793         yield conn
    794     except Exception as e:

c:\users\jenielse\source\repos\qcodes\qcodes\dataset\sqlite_base.py in atomic_transaction(conn, sql, *args)
    755     with atomic(conn) as conn:
--> 756         c = transaction(conn, sql, *args)
    757     return c

c:\users\jenielse\source\repos\qcodes\qcodes\dataset\sqlite_base.py in transaction(conn, sql, *args)
    732     else:
--> 733         c.execute(sql)
    734     return c

OperationalError: near "FROM": syntax error

The above exception was the direct cause of the following exception:

RuntimeError                              Traceback (most recent call last)
<ipython-input-12-4bf7cb4ff8cf> in <module>
----> 1 datasaver.dataset.get_parameter_data()

c:\users\jenielse\source\repos\qcodes\qcodes\dataset\data_set.py in get_parameter_data(self, start, end, *params)
    768         valid_param_names = self._validate_parameters(*params)
    769         return get_parameter_data(self.conn, self.table_name, valid_param_names,
--> 770                         start, end)
    771 
    772     def get_values(self, param_name: str) -> List[List[Any]]:

c:\users\jenielse\source\repos\qcodes\qcodes\dataset\sqlite_base.py in get_parameter_data(conn, table_name, columns, start, end)
   1161     # This method is going to form the entry point for a compiled version
   1162 
-> 1163     res = get_data(conn, table_name, columns, start, end)
   1164     res_t = list(map(list, zip(*res)))
   1165     return {name: np.array(column_data)

c:\users\jenielse\source\repos\qcodes\qcodes\dataset\sqlite_base.py in get_data(conn, table_name, columns, start, end)
   1132 
   1133     query = _build_data_query(table_name, columns, start, end)
-> 1134     c = atomic_transaction(conn, query)
   1135     res = many_many(c, *columns)
   1136 

c:\users\jenielse\source\repos\qcodes\qcodes\dataset\sqlite_base.py in atomic_transaction(conn, sql, *args)
    754     """
    755     with atomic(conn) as conn:
--> 756         c = transaction(conn, sql, *args)
    757     return c
    758 

~\AppData\Local\Continuum\miniconda3\envs\qcodes\lib\contextlib.py in __exit__(self, type, value, traceback)
     97                 value = type()
     98             try:
---> 99                 self.gen.throw(type, value, traceback)
    100             except StopIteration as exc:
    101                 # Suppress StopIteration *unless* it's the same exception that

c:\users\jenielse\source\repos\qcodes\qcodes\dataset\sqlite_base.py in atomic(conn)
    795         conn.rollback()
    796         log.exception("Rolling back due to unhandled exception")
--> 797         raise RuntimeError("Rolling back due to unhandled exception") from e
    798     else:
    799         if is_outmost:

RuntimeError: Rolling back due to unhandled exception


```